requests/utils.py

```diff
@@ -665,6 +665,21 @@ def parse_header_links(value):
 
     return links
 
+def is_valid_location(response):
+    """Verify that multiple Location headers weren't
+    returned from the last response.
+    """
+    if hasattr(response.raw, 'headers'):
+        if hasattr(response.raw.headers, 'getlist'):
+            return len(response.raw.headers.getlist('location')) <= 1
+        else:
+            pass # Do something with header value
+    elif hasattr(response.raw, 'read'):
+        response.raw.seek(0) # Move to beginning of file
+        data = response.raw.read()
+        # parse_headers(data) # try to parse headers out of raw response
```

Comment:
Ok, great, that helps narrow the scope considerably. I'll shorten this down to just catching the urllib3 case then. Thanks!